cpp/ql/src/experimental/Best Practices/WrongUintAccess.ql

```diff
@@ -0,0 +1,23 @@
+/**
+ * @id cpp/wrong-uint-access
+ * @name Wrong Uint
+ * @descripion Acess an array of size lower than 256 with a uint16.
+ * @kind problem
+ * @problem.severity recommendation
+ * @tags efficiency
+ */
+
+import cpp
+import semmle.code.cpp.controlflow.SSA
+
+from
+  Variable E, ArrayExpr useExpr, ArrayType defExpr, VariableDeclarationEntry def, VariableAccess use
+where
+  def = defExpr.getATypeNameUse() and
+  E = def.getDeclaration() and
```

Comment:
`getATypeNameUse` is really a best-effort predicate that likely won't give you all the cases you're interested in. As far as I can tell, you're really only interested in knowing that `def` is an `ArrayExpr` with some size, right? In that case, I think you could just do: `E.getType() = defExpr`. Instead of using `getType()` you could also use `getUnspecifiedType()` to automatically handle typedefs and `const`/`volatile` specifiers.

Reply:
Done

Reply:
Awesome. Thanks a lot!